fieldservice_sale/__manifest__.py

```diff
@@ -11,6 +11,7 @@
     'depends': [
         'fieldservice',
         'sale_management',
+        'fieldservice_account',
```

Comment:
Please don't add this dependency.

Reply:
@hparfr I agree adding the dependency adds a lot of extra logic from fieldservice_account that is not needed for this addition, but if this dependency is not used, there is no `fsm_order_id` field for `account.invoice` model which we set here from the `sale_order`

Thoughts?

Reply:
@hparfr `sale_management` depends on `account`.

If a FSM product is set to be invoiced on quantity delivered, we need the ability to invoice the completed FSM order.

Reply:
Yes, but fieldservice_account take strong assumption about how invoicing should be done
and some dependencies (analytic, hr_timesheet)

Reply:
How would you do it?

Reply:
Can we split fieldservice_account so that the way it does invoicing is in a separate module and keep only things like adding fsm_order_id to the account.invoice model?
I don't mind doing the work

Reply:
@max3903 @hparfr Take a look at #411 and please provide your feedback

Reply:
Thanks for this work @brian10048.
I still think billing_id (customer_id) and costs should not be in fsm_order. There is not so much stuff left in fieldservice_accounting.

IMO fieldservice_sale should: 
- link fsm_orders to sale_orders (ok)
- choose how fsm_orders are created (ok field_service_tracking)
- create fsm_orders from sale (ok)
- link fsm_orders to account_invoice_lines (1)
- compute invoiceable quantity (ok)
- prepare invoice lines (ok)
- add links and smart buttons to navigate from invoice to fsm orders, vice versa.
- do not depend of fieldservice_account

1): About link between account_invoice_line and fsm_order:
* An FSM order should be invoiced once.
* Different FSM Orders can be on the same invoice line.

Reply:
@hparfr I am fine with this. On the last 2 items (compute invoiceable quantity and prepare invoice lines), if you invoice on delivered quantity, you would invoice from the sale order, right? Not from the FSM order (like inventory items are not invoiced from the delivery order).

fieldservice_sale --> Invoice the service from the sale order before or after the completion of the service (--> no timesheet, no cost)
fieldservice_account --> Invoice timesheet/costs from the FSM order
fieldservice_sale_account --> Invoice the service + timesheet/costs from either the FSM order or the sale order, before (no timesheet/cost) or after the completion of the service

Reply:
Indeed @max3903 it's like sale orders and pickings:  _compute_qty_delivered() is currently in fieldservice_sale/models/sale_order_line.py

In the actual proposal
- fieldservice_account: links a fsm_orders to invoices lines. Kind of base module.
- fieldservice_account_analytic:  add timesheet, costs, billing_id, and so on. Depends of fieldservice_account
- fieldservice_sales, create fsm orders at sale confirmation and set quantities when invoice is created from sale. Depends of fieldservice_account
- fieldservice_contract, set quantities when invoice is created from contract. Depends of fieldservice_account

Reply:
:+1:

Reply:
@hparfr @brian10048 See #415 where I copied `fieldservice_account` and removed the `fsm_order_id` field on `account.invoice.line`

Reply:
@max3903  have you seen https://github.com/brian10048/field-service/pull/8 ? 
I will look at #415 in few days.

Reply:
I will be able to look at all of this over the next few days as well and provide feedback on all of the above

Reply:
Resolved with #411